src/tests/integration/test-data/symbolic-bytes-lemmas.k

```diff
@@ -46,4 +51,36 @@ module SYMBOLIC-BYTES-LEMMAS
       requires N <=Int lengthBytes(B)
       [simplification(60), concrete(B, N)]
 
+    //
+    // String-related manipulation
+    //
+    rule [bitwise-and-to-right]:
+      X:Int &Int #asWord ( B1:Bytes +Bytes B2:Bytes ) => X &Int #asWord ( B2 )
+      requires #rangeUInt(256, X)
+       andBool lengthBytes(B1:Bytes +Bytes B2) <=Int 32
+       andBool X +Int 1 ==Int 2 ^Int log2Int(X +Int 1)
+       andBool log2Int(X +Int 1) <=Int lengthBytes(B2) *Int 8
+       [simplification, concrete(X), preserves-definedness]
+
+    rule [bitwise-and-shift]:
+      X:Int &Int ( #asWord ( B1:Bytes +Bytes B2:Bytes ) >>Int Y:Int ) => X &Int ( #asWord ( B2 ) >>Int Y )
+      requires #rangeUInt(256, X)
+       andBool 0 <=Int Y andBool Y <Int 256
+       andBool lengthBytes(B1:Bytes +Bytes B2) <=Int 32
+       andBool X +Int 1 ==Int 2 ^Int log2Int(X +Int 1)
+       andBool log2Int(X +Int 1) <=Int ( lengthBytes(B2) *Int 8 -Int Y )
+       [simplification, concrete(X, Y), preserves-definedness]
+
+    rule [string-is-short]:
+      1 &Int ( 2 *Int X:Int ) => 0
+      requires 0 <=Int X
+      [simplification]
+
+    //
+    // Arithmetic
+    //
+    rule ( A:Int *Int B:Int ) >>Int C:Int => ( A /Int ( 2 ^Int C ) ) *Int B
```

Comment:
Do we want nonnegative constraints for any of `A`/`B`/`C`?

Reply:
Yes, we certainly do.